### PR TITLE
add spacer element to statusline

### PR DESCRIPTION
**Pull request by sbromberger**
Addresses #3106 and associated other comments. Multiple `"spacer"` elements may be specified in each statusline section.

### CODE REVIEW

**Comment by A-Walrus** [comment]
This would probably need to be added to the documentation (`book/src/configuration.md:80`)

**Comment by sbromberger** [comment]
> This would probably need to be added to the documentation (`book/src/configuration.md:80`)

Of course. Sorry for the oversight. Will update.

**Comment by sbromberger** [comment]
Ready for review.

**Comment by AceofSpades5757** [comment]
Currently, gutters uses `"padding"`, would that not be more appropriate terminology compared to `"spacing"`?

[Helix Master Docs](https://docs.helix-editor.com/master/configuration.html?highlight=padding#editor-section)

File: helix-view/src/editor.rs [hunk 1]
```diff
@@ -246,6 +246,9 @@ pub enum StatusLineElement {
 
     /// The cursor position
     Position,
+
+    /// A single space
+    Spacer,
```
Comment (AceofSpades5757) [inline helix-view/src/editor.rs:251]: `GutterType` uses `Padding` terminology.
`helix-view/src/editor.rs` on line 330.

File: book/src/configuration.md [hunk 2]
```diff
@@ -78,6 +78,7 @@ The following elements can be configured:
 | `diagnostics` | The number of warnings and/or errors |
 | `selections` | The number of active selections |
 | `position` | The cursor position |
+| `spacer` | Inserts a space between elements (multiple/contiguous spacers may be specified) |
```
Comment (AceofSpades5757) [inline book/src/configuration.md:81]: Gutters uses `"padding"`.

`/book/src/configuration.md` on line 41.

```
| `gutters` | Gutters to display: Available are `diagnostics` and `line-numbers` and `padding`, note that `diagnostics` also includes other features like breakpoints | `["diagnostics", "line-numbers", "padding"]` |
```

**Comment by sbromberger** [comment]
To me, "padding" implies inserting spaces at the front (or back) of strings to align columns. That's not what these do, but I'm happy to change the name if there's a consensus.

File: helix-view/src/editor.rs (hunk 1 above)
Comment (archseer) [reply helix-view/src/editor.rs:251]: Gutters should also use the spacer terminology, I'll rename there

**Pull request merged**